src/serializers/extra.rs

```diff
@@ -424,8 +425,19 @@ impl CollectWarnings {
                 .get_type()
                 .qualname()
                 .unwrap_or_else(|_| PyString::new_bound(value.py(), "<unknown python object>"));
+
+            let input_str = safe_repr(value);
+            let mut value_str = String::with_capacity(100);
+            value_str.push_str("with value `");
+            match crate::errors::truncate_large_string(&mut value_str, input_str.to_cow()) {
```

Comment:
is `Write` for `String` just infallible and we can `unwrap` here :thinking:

Reply:
Looks like it: https://doc.rust-lang.org/1.80.0/src/alloc/string.rs.html#2918

Reply:
Though maybe use `.expect()`? I tend to like to justify why; the message should be optimized away if the function is truly infallible right? 😂